src/CLI/Options.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CommandLine;
+using System;
+using System.IO;
+
+namespace AxeWindowsScanner
+{
+    public class Options : IOptions
+    {
+        [Option(Required = false, HelpText = "Output directory")]
+        public string OutputDirectory => _outputDirectory;
+
+        [Option(Required = false, HelpText = "Scan ID")]
+        public string ScanId => _scanId;
+
+        [Option(Group = "Target", Required = false, HelpText = "Process Id")]
+        public int ProcessId => _processId;
+
+        [Option(Group = "Target", Required = false, HelpText = "Process Name")]
+        public string ProcessName => _processName;
+
+        [Option(Required = false, HelpText = "Verbosity level (Quiet/Default/Verbose)")]
+        public string Verbosity => string.Empty;  // Exists only for CommandLine purposes
+
+        public VerbosityLevel VerbosityLevel { get; }
+
+        // We don't control construction of this class, so we use static Dependency Injection
+        public static IErrorCollector ErrorCollector { get; set; }
+        public static IProcessHelper ProcessHelper { get; set; }
+
+        readonly string _outputDirectory;
+        readonly string _scanId;
+        readonly int _processId;
+        readonly string _processName;
+
+        public Options(string outputDirectory, string scanId, int processId, string processName, string verbosity)
+        {
+            _outputDirectory = outputDirectory;
+            _scanId = scanId;
+            VerbosityLevel = VerbosityLevel.Default;  // Until proven otherwise
+
+            if (!string.IsNullOrEmpty(verbosity))
+            {
+                if (Enum.TryParse<VerbosityLevel>(verbosity, true, out VerbosityLevel level))
+                {
+                    VerbosityLevel = level;
+                }
+                else
+                {
+                    ErrorCollector.AddParameterError("Invalid verbosity level: " + verbosity);
+                }
+            }
+
+            if (processId != 0)
+            {
+                _processId = processId; 
```

Comment:
Thought: Should the options class be doing this? I kind of feel like the options class should be meant to simply contain the options and not have much program logic.

Reply:
Options knows about the input parameters, then the rest of the program treats the options as immutable. I went back and forth a couple of times on how to best use this. Happy to talk offline...